src/adapters/github.js

```diff
@@ -149,10 +153,98 @@ class GitHub extends PjaxAdapter {
   _getTree(path, opts, cb) {
     this._get(`/git/trees/${path}`, opts, (err, res) => {
       if (err) cb(err)
-      else cb(null, res.tree)
+      else {
+        if (!opts.repo.pull) cb(null, res.tree)
+        else {
+          this._getPatch(opts, (patchErr, patchRes) => {
+            const diffExists = patchRes && Object.keys(patchRes).length > 0
+            if (patchErr || !diffExists) cb(null, res.tree)
+            else {
+              // Filter tree to only include files and directories that are included in the patch
+              const filteredTree = res.tree
```

Comment:
Another bug I just caught with this code:
If a file is added or renamed, it doesn't exist in `res.tree`.  This means these files won't show up in the tree view.

To fix this, we will only use the result from `_getPatch` and return an array with objects that are similarly structured to those returned by the `/git/trees/${path}` call.  This has the added benefit of only making one network call rather than two as it is now.